src/feature_builder.hpp

```diff
@@ -220,49 +223,70 @@ struct overzoomed_feature_builder
     {
         std::vector<detail::annotated_ring<CoordinateType>> rings;
         vtzero::decode_polygon_geometry(feature.geometry(), detail::polygon_handler<CoordinateType>(rings, dx_, dy_, zoom_factor_));
-        if (!rings.empty())
+        if (rings.empty())
         {
-            vtzero::polygon_feature_builder feature_builder{layer_builder_};
-            feature_builder.copy_id(feature);
-            bool valid = false;
-            bool process = false;
-            for (auto& r : rings)
+            return;
+        }
+        mapbox::geometry::multi_polygon<coordinate_type> results;
+        mapbox::geometry::wagyu::wagyu<std::int64_t> wagyu;
+        bool data_added = false;
```

Comment:
It seems like it might be clearer to combine `data_added` and `!skip_inner` as a single `added_outer_ring` condition.